From a9e08b30cfa91ddfd57176ffc43f2e6f02292a31 Mon Sep 17 00:00:00 2001
From: Dmitry Shifrin <dmitry.shifrin@cogentembedded.com>
Date: Thu, 8 Sep 2016 20:46:37 +0300
Subject: [PATCH 190/433] media: soc_camera: rcar_vin: Check width align

We have to check cam->out_width align because it is a requirement of
VnIS register.

Signed-off-by: Dmitry Shifrin <dmitry.shifrin@cogentembedded.com>
Signed-off-by: Nguyen Van Linh [FGA.BU13] <LinhNV25@fsoft.com.vn>
Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 drivers/media/platform/soc_camera/rcar_vin.c | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/drivers/media/platform/soc_camera/rcar_vin.c b/drivers/media/platform/soc_camera/rcar_vin.c
index dfb2c72..e613d0b 100644
--- a/drivers/media/platform/soc_camera/rcar_vin.c
+++ b/drivers/media/platform/soc_camera/rcar_vin.c
@@ -1663,10 +1663,6 @@ static int rcar_vin_set_fmt(struct soc_camera_device *icd,
 	if (mf.code != xlate->code)
 		return -EINVAL;
 
-	/* Prepare VIN crop */
-	cam->width = mf.width;
-	cam->height = mf.height;
-
 	/* Use VIN scaling to scale to the requested user window. */
 
 	/* We cannot scale up */
@@ -1683,6 +1679,14 @@ static int rcar_vin_set_fmt(struct soc_camera_device *icd,
 		pix->height = vin_sub_height;
 	}
 
+	if (can_scale && !IS_ALIGNED(pix->width, 0x10)) {
+		return -EINVAL;
+	}
+
+	/* Prepare VIN crop */
+	cam->width = mf.width;
+	cam->height = mf.height;
+
 	/*
 	 * We have calculated CFLCR, the actual configuration will be performed
 	 * in rcar_vin_set_bus_param()
-- 
2.7.4

